diff -ruN cbmbasic-1.0/cbmbasic.c cbmbasic-1.0.aros/cbmbasic.c
--- cbmbasic-1.0/cbmbasic.c	2009-04-08 04:13:13.000000000 +0000
+++ cbmbasic-1.0.aros/cbmbasic.c	2021-01-17 23:45:06.540000000 +0000
@@ -137,19 +137,6 @@
 
 
 /* Global Variable Declarations */
-extern unsigned char A;
-extern unsigned char RAM[65536];
-extern unsigned short PC;
-extern unsigned char S;
-extern unsigned char X;
-extern unsigned char Z;
-extern unsigned char N;
-extern unsigned char C;
-extern unsigned char V;
-extern unsigned char Y;
-extern unsigned char B;
-extern unsigned char D;
-extern unsigned char I;
 static unsigned short _ZZ4mainE11case_labels[17591];
 
 
diff -ruN cbmbasic-1.0/runtime.c cbmbasic-1.0.aros/runtime.c
--- cbmbasic-1.0/runtime.c	2009-04-08 03:00:03.000000000 +0000
+++ cbmbasic-1.0.aros/runtime.c	2021-01-17 23:44:14.120000000 +0000
@@ -42,7 +42,7 @@
 #include "plugin.h"
 #include "glue.h"
 
-unsigned char RAM[65536];
+extern unsigned char RAM[65536];
 
 int
 stack4(unsigned short a, unsigned short b, unsigned short c, unsigned short d) {
@@ -390,7 +390,7 @@
 			putchar(13);
 			putchar(10);
 			break;
-#ifdef _WIN32
+#if defined(_WIN32) || defined(__AROS__)
 		case 0x1D: /* CSR RIGHT */
 			putchar(' ');
 			break;
@@ -608,6 +608,8 @@
 	st.wSecond       = (WORD)(seconds%60);
 	st.wMilliseconds = (WORD)((jiffies % 60) * 1000 / 60);
 	SetLocalTime(&st);
+#elif __AROS__
+    #warning SETTIM not implemented because of missing settimeofday()
 #else
     time_t  now = time(0);
     struct tm       bd;
